### PR TITLE
[Win32] Terminate agent-manager.exe on uninstall/upgrade.

### PR DESCRIPTION
**Pull request by derekwbrown**
### What does this PR do?

Modifies the installer so that if agent-manager.exe is running, it is automatically terminated during uninstall and/or upgrade.  This prevents requiring a reboot after upgrade.

### Motivation

Multiple failed installs due to agent-manager.exe running in the background.

### Additional Notes

Anything else we should know when reviewing?

Modifies install script to terminate agent-manager during uninstall or
upgrade.  Otherwise, agent-manager is running in the background and can't
be upgraded, leading to an unstable install

### CODE REVIEW

File: win32/gui.py [hunk 1]
```diff
@@ -845,6 +845,10 @@ def kill_old_process():
     if Platform.is_windows():
         # Let's kill any other running instance of our GUI/SystemTray before starting a new one.
         kill_old_process()
+        if len(sys.argv) > 1 and "-stop" in sys.argv:
+            # just return.  The kill_old_process() should have terminated the process,
+            # and now we're done.
+            sys.exit(0)
```
Comment (yannmh) [inline win32/gui.py]: I don't understand what this is for.

Comment (derekwbrown) [inline win32/gui.py]: I can take it out if need be. Originally, I thought I was going to need to "shell out" from the installer (in Windows parlance, a Custom Action) to stop the agent-manager if it's running.  Turns out I didn't need to.  However, it seems potentially useful to be able to stop it from the command line.  There'd be no harm removing.

Comment (yannmh) [inline win32/gui.py]: So
```
agent-manager.exe -stop
```
should stop the background process ?

Comment (derekwbrown) [inline win32/gui.py]: Yes.

**Comment by yannmh** [comment]
Let's merge it 🚀

**Pull request merged**